watertap/unit_models/tests/unit_test_harness.py

```diff
@@ -167,79 +167,82 @@ def test_unit_solution(self, frame_unit):
-                stream = getattr(blk, key[0])  # m.fs.feed_side
-                var = getattr(stream[0], key[1])[
-                    key[2]
-                ]  # m.fs.feed_side.properties_in.flow_mass_phase_comp[Liq, NaCl]
-                # relative tolerance doesn't mean anything for 0-valued things
-                if val == 0:
-                    if not pytest.approx(val, abs=1.0e-08) == value(var):
-                        raise UnitValueError(
-                            "Variable {v_name} is expected to have a value of {val} +/- 1.0e-08, but it "
-                            "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-                            "that sets up the UnitRegressionTest".format(
-                                v_name=key[1], ind=key[2], val=val, val_t=value(var)
-                            )
-                        )
-                elif not pytest.approx(val, rel=1e-3) == value(var):
-                    raise UnitValueError(
-                        "Variable {v_name} is expected to have a value of {val} +/- 0.1%, but it "
-                        "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-                        "that sets up the UnitRegressionTest".format(
-                            v_name=key[1], ind=key[2], val=val, val_t=value(var)
-                        )
-                    )
-
-            elif len(key) == 2:
-                # for (v_name, ind), val in solution.keys():
-                var = getattr(blk, key[0])[
-                    key[1]
-                ]  # m.fs.feed_side.properties_in.flow_mass_phase_comp[Liq, NaCl]
-                # relative tolerance doesn't mean anything for 0-valued things
-                if val == 0:
-                    if not pytest.approx(val, abs=1.0e-08) == value(var):
-                        raise UnitValueError(
-                            "Variable {v_name} is expected to have a value of {val} +/- 1.0e-08, but it "
-                            "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-                            "that sets up the UnitRegressionTest".format(
-                                v_name=key[0], ind=key[1], val=val, val_t=value(var)
-                            )
-                        )
-                elif not pytest.approx(val, rel=1e-3) == value(var):
-                    raise UnitValueError(
-                        "Variable {v_name} is expected to have a value of {val} +/- 0.1%, but it "
-                        "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-                        "that sets up the UnitRegressionTest".format(
-                            v_name=key[0], ind=key[1], val=val, val_t=value(var)
-                        )
-                    )
+        for key, val in solutions.items():
+            pyo_obj = eval(key, {"m": self.model})
```

Comment:
I'm not sure what's the relationship between the `UnitTestHarness` instance (i.e. `self` in this scope) and `blk._test_objs`.

If attributes set in the body of the model-specific subclasses are supposed to be addressed via `blk._test_objs`, then you might need to use `{"m": blk._test_objs.model}` in the second argument of the `eval()` call.

Reply:
If this still doesn't work, I might have to take a more in-depth look into this - there are a number of subtleties involved when using pytest `Test` classes (essentially, pytest only uses them as namespaces for test functions and most of the usual class instance functionality doesn't work as one would expect. See https://docs.pytest.org/en/stable/getting-started.html#group-multiple-tests-in-a-class for more info).

Reply:
Thanks for the help @lbianchi-lbl! I'll take a look at these at some point tomorrow.